### ISSUE TITLE
Windows binary

### ISSUE THREAD

**Issue by burtek** [l1ving/youtube-dl#22]
Any chance for windows binary?

**Comment by l1ving** [comment]
Well, if I knew how to build a windows binary I'd release one. The docs don't describe it and as I said in #2, I'm not an official youtube-dl maintainer. Hopefully someone can point me to how I can do that, I'll ask on the irc later today

**Comment by solbu** [comment]
@burtek: The official site https://yt-dl.org/ has just released a new version which also has a windows binary, so you might want to use that one.

**Comment by BachoSeven** [comment]
@solbu do you think they are continuing development in private or is there already an alternative git platform where dev is going on?

**Comment by l1ving** [comment]
Well I wonder what the official site changed because there's not git from what I can see.

**Comment by burtek** [comment]
@solbu thanks for info! I checked it just before posting here and there was no update.

**Comment by samoht0** [comment]
> Well I wonder what the official site changed because there's not git from what I can see.

Temporary "official" youtube-dl repository on gitlab:
https://gitlab.com/dstftw/youtube-dl

**Comment by l1ving** [comment]
Thank you! I've backported their changes

**Comment by Ash-png** [comment]
You can use py2exe "compile" python code
youtube-dl even have setup.py just for that
try: `python .\setup.py py2exe` in root dir.
remember install missing Modules

py2exe: [http://www.py2exe.org/index.cgi/Tutorial](url)

**Comment by Ash-png** [comment]
Sorry for my bad English.
I delete my previous comment because it offer no help and retarded.
Here is how to build youtube-dl.exe on window (not official).

1. install python3 (I use Python 3.7 provide by Visual Studio)
2. make sure python's environment variables is set
Example :
Fallowing add to `Path` environment variables
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\Scripts
Fallowing add to `PYTHONPATH` environment variables
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\Lib

3. open powerShell and cd to youtube-dl's root project directory (where setup.py reside)
4. ` python -m pip install py2exe`
5. you can type `python setup.py --help-commands` to see build option
6.  ` python setup.py build`
7.  ` python setup.py install` (require administrator)
8.  ` python setup.py py2exe` (It is fine if it show "missing module", It is just warning message)
9. If everything go will you can see youtube-dl.exe (and some *.dll in my case) in youtube-dl's project root directory

note
1. do not use python from window app store
2. this is not official way to build youtube-dl.exe , original team maybe using wine-py2exe.sh (in youtube-dl/devscripts) under linux.
3. I am not Python developer but I can aid small help like this, wish I can contribute in the future after familiar with python.
4. fcuk RIAA and its abuse of power

**Comment by PieterjanDeClippel** [comment]
> fcuk RIAA and its abuse of power

The RIAA doesn't have any power. If they want to counter something, they have to go through a court.

The same goes for GitHub. They should have never removed everything without a court order, hence why GitHub is an unreliable party for hosting git-repositories.

**Comment by Project-Magenta** [comment]
It was a case-and-desisit

**Comment by Aractus** [comment]
The codebase is just hidden from the public, you can still update using `youtube-dl.exe -U`, and you can still install the python version as well `py -m pip install youtube-dl` or `py -m pip install --upgrade youtube-dl`.

**Comment by l1ving** [comment]
That's because the pip repos didn't take it down

**Comment by Aractus** [comment]
Right, nor were they asked to - https://pypi.org/simple/youtube-dl is not in the DMCA notice. I'm just noting that there's no disruption to getting the latest version at all.
